Makefile.kubebuilder

```diff
@@ -0,0 +1,127 @@
+manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
+	$(CONTROLLER_GEN) rbac:roleName=manager-role crd webhook paths="github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1" output:crd:artifacts:config=pkg/customresource
+
+.PHONY: generate
+generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
+	$(CONTROLLER_GEN) object:headerFile="pkg/hack/boilerplate.go.txt" paths="github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+
+.PHONY: fmt
+fmt: ## Run go fmt against code.
+	go fmt ./...
+
+.PHONY: vet
+vet: ## Run go vet against code.
+	go vet ./...
+
+##@ Build
+
+.PHONY: kubebuilder-build
+kubebuilder-build: generate fmt vet ## Build manager binary.
+	go build -o bin/manager ./cmd/reposervercontroller/main.go
+
+.PHONY: kubebuilder-run
+kubebuilder-run: manifests generate fmt vet ## Run a controller from your host.
+	go run ./cmd/reposervercontroller/main.go
+
+# If you wish built the manager image targeting other platforms you can use the --platform flag.
+# (i.e. docker build --platform linux/arm64 ). However, you must enable docker buildKit for it.
+# More info: https://docs.docker.com/develop/develop-images/build_enhancements/
+.PHONY: docker-build
+docker-build: kubebuilder-test ## Build docker image with the manager.
+	docker build -t ${IMG} docker/repositoryserver-controller
+
+.PHONY: docker-push
+docker-push: ## Push docker image with the manager.
+	docker push ${IMG}
+
+# PLATFORMS defines the target platforms for  the manager image be build to provide support to multiple
+# architectures. (i.e. make docker-buildx IMG=myregistry/mypoperator:0.0.1). To use this option you need to:
+# - able to use docker buildx . More info: https://docs.docker.com/build/buildx/
+# - have enable BuildKit, More info: https://docs.docker.com/develop/develop-images/build_enhancements/
+# - be able to push the image for your registry (i.e. if you do not inform a valid value via IMG=<myregistry/image:<tag>> than the export will fail)
+# To properly provided solutions that supports more than one platform you should use this option.
+PLATFORMS ?= linux/amd64
+.PHONY: docker-buildx
+docker-buildx: kubebuilder-test ## Build and push docker image for the manager for cross-platform support
+	# copy existing Dockerfile and insert --platform=${BUILDPLATFORM} into Dockerfile.cross, and preserve the original Dockerfile
+	sed -e '1 s/\(^FROM\)/FROM --platform=\$$\{BUILDPLATFORM\}/; t' -e ' 1,// s//FROM --platform=\$$\{BUILDPLATFORM\}/' docker/repositoryserver-controller/Dockerfile > docker/repositoryserver-controller/Dockerfile.cross
+	- docker buildx create --name project-v3-builder
+	docker buildx use project-v3-builder
+	- docker buildx build --push --platform=$(PLATFORMS) --tag ${IMG} -f docker/repositoryserver-controller/Dockerfile.cross
+	- docker buildx rm project-v3-builder
+	rm docker/repositoryserver-controller/Dockerfile.cross
```

Comment:
Does it make sense to call these targets from the existing `build`, `build-controller` and `release` targets. Not right now but once we have everything ready. Also, let's have some prefixes in these reposerver-specific targets to differentiate them from existing ones. Maybe `reposerver-*` or something like that